ext/openssl/openssl.c

```diff
@@ -5948,17 +6042,20 @@ PHP_FUNCTION(openssl_cms_decrypt)
 	char * outfilename;
 	size_t outfilename_len;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "ppz|zl", &infilename, &infilename_len,
-				&outfilename, &outfilename_len, &recipcert,
-				&recipkey, &encoding) == FAILURE) {
-		RETURN_THROWS();
-	}
+	ZEND_PARSE_PARAMETERS_START(3, 5)
+		Z_PARAM_PATH(infilename, infilename_len)
+		Z_PARAM_PATH(outfilename, outfilename_len)
+		Z_PARAM_ZVAL(recipcert)
```

Comment:
Wouldn't it be possible to use X509|string here too?
```suggestion
		Z_PARAM_STR_OR_OBJ_OF_CLASS(recipcert_str, recipcert_object, x509_certificate_ce)
```
Otherwise I don't see why this is using FastZPP as it doesn't make any changes.

Reply:
Unfortunately, the problem is that a few lines below, `recipcert` is used in `php_openssl_evp_from_zval()` which handles arrays as well. :(